src/commands/cicdPipeline.ts

```diff
@@ -0,0 +1,64 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import * as utils from 'util';
+import AksClusterTreeItem from '../tree/aksClusterTreeItem';
+import { IActionContext } from 'vscode-azureextensionui';
+
+const deployToAzureExtensionId = 'ms-vscode-deploy-azure.azure-deploy';
+const configurePipelineCommand = 'configure-cicd-pipeline';
+const browsePipelineCommand = 'browse-cicd-pipeline';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void> {
+    if (await isDeployToAzureExtensionInstalled()) {
+        const cluster = await getCluster(target);
+        if (!!cluster) {
+            await executeDeployToAzureExtensionInstalled(configurePipelineCommand, cluster);
+        }
+    }
+}
+
+export async function browsePipeline(context: IActionContext, target: any): Promise<void> {
+    if (await isDeployToAzureExtensionInstalled()) {
+        const cluster = await getCluster(target);
+        if (!!cluster) {
+            await executeDeployToAzureExtensionInstalled(browsePipelineCommand, cluster);
+        }
+    }
+}
+
+async function getCluster(target: any): Promise<AksClusterTreeItem | undefined> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    if (cloudExplorer.available) {
+        const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+        if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+            return clusterTarget.cloudResource as AksClusterTreeItem;
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
```

Comment:
will we ever land here as in the manifest, you are specifying a condition of `viewItem =~ /aks\\.cluster/i`

Reply:
But this can come here if the cloudProvider isn't Azure.